// Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";
package baidu.paddle_serving.pyserving;

service GeneralPythonService {
  rpc inference(Request) returns (Response) {}
}

message Request {
  repeated bytes feed_insts = 1;
  repeated string feed_var_names = 2;
  repeated bytes shape = 3;
  repeated string type = 4;
}

message Response {
  repeated bytes fetch_insts = 1;
  repeated string fetch_var_names = 2;
  required int32 ecode = 3;
  optional string error_info = 4;
  repeated bytes shape = 5;
  repeated string type = 6;
}
